### PR TITLE
Feature/essence descriptor equivalence

### PR DESCRIPTION
**Pull request by schakrovorthy**

### CODE REVIEW

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java [hunk 1]
```diff
@@ -53,6 +57,44 @@ public static List checkVirtualTracks(IMFCompositionPlaylistType compositionPlay
...
+            {
+                //Construct a DOMNodeObjectModel object for every EssenceDescriptor that is a part of the EssenceDescriptorList
+                Map<UUID, List<DOMNodeObjectModel>> domNodeObjectModelMap = new HashMap<>();
+                List<IMFEssenceDescriptorBaseType> essenceDescriptors = compositionPlaylistType.getEssenceDescriptorList();
+                for(IMFEssenceDescriptorBaseType essenceDescriptorBaseType : essenceDescriptors){
+                    List<DOMNodeObjectModel> domNodeObjectModels = new ArrayList<>();
+                    for (Object object : essenceDescriptorBaseType.getAny()) {
+                        domNodeObjectModels.add(new DOMNodeObjectModel((Node) object));
+                    }
+                    domNodeObjectModelMap.put(essenceDescriptorBaseType.getId(), domNodeObjectModels);
```
Comment (svenkatrav) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java]: getEssenceDescriptorListMap() already implements this.

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java [hunk 2]
```diff
@@ -53,6 +57,44 @@ public static List checkVirtualTracks(IMFCompositionPlaylistType compositionPlay
...
+                }
+
+                if(!(virtualTrackEssenceDescriptors.size() > 0)){
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CPL_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.FATAL, String.format("This Composition represented by the ID %s is invalid since the VirtualTrack represented by ID %s doesn't seem to refer to a single EssenceDescriptor in the CPL's EssenceDescriptorList", compositionPlaylistType.getId().toString(), virtualTrack.getTrackID().toString()));
+                }
+                boolean isVirtualTrackHomogeneous = true;
+                DOMNodeObjectModel refDOMNodeObjectModel = virtualTrackEssenceDescriptors.get(0);
+                for(int i=1; i<virtualTrackEssenceDescriptors.size(); i++){
+                    isVirtualTrackHomogeneous &= refDOMNodeObjectModel.equivalent(virtualTrackEssenceDescriptors.get(i));
+                    refDOMNodeObjectModel = virtualTrackEssenceDescriptors.get(i);
```
Comment (svenkatrav) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java]: I think we need not update the ref.

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java (hunk 1 above)
Comment (schakrovorthy) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java]: getEssenceDescriptorListMap() is implemented in Composition guess I could pass in the Composition object to the IMFCoreConstraintsChecker() I just took the approach of not changing the interface to checkVirtualTracks() and construct the list internally.

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java (hunk 2 above)
Comment (schakrovorthy) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java]: Yes, we needn't however by doing this if we need to debug this code we will get to know if the first EssenceDescriptor is the odd one or not.

**Pull request merged**